pkgs/tools/networking/autossh/default.nix

```diff
@@ -2,14 +2,19 @@
 
 stdenv.mkDerivation rec {
   name = "autossh-1.4g";
-  
+
   src = fetchurl {
     url = "http://www.harding.motd.ca/autossh/${name}.tgz";
     sha256 = "0xqjw8df68f4kzkns5gcah61s5wk0m44qdk2z1d6388w6viwxhsz";
   };
-  
-  buildInputs = [ openssh ];
-  
+
+  preConfigure = ''
+    export ac_cv_func_malloc_0_nonnull=yes
+    export ac_cv_func_realloc_0_nonnull=yes
```

Comment:
This claim is false and potentially dangerous on some platforms. My guess is that `malloc(0)` will return a valid non-`NULL` address in most new `libc`s.
So does autossh depend on this behavior or is this check just collateral damage of using autotools?

Reply:
A safer solution is to just define `rpl_malloc` and `rpl_realloc`.

Reply:
this is the clearest documentation I could find on it: https://web.archive.org/web/20120302062736/http://wiki.buici.com:80/xwiki/bin/view/Programing+C+and+C%2B%2B/Autoconf+and+RPL_MALLOC
do you want me to add a patch to `autossh.c` that defines these?
last note: I don't believe that autossh depends on this functionality.

Reply:
oh we can follow the yocto people: https://github.com/olewolf/meta-olewolf/blob/master/recipes-connectivity/autossh/files/01fix-rpl-malloc.patch

Reply:
It's a tiny program. The allocation calls seem safe indeed.

I don't think the yocto patch is any better than your solution. Both ignore the test and are potentially unsafe in general / in the future. I don't think autossh will start depending on that behavior soon, so the chances are very low for this to be a problem.